### PR TITLE
Fix for download problem by not adding parameter user_catalog_file to analysis_parameters.json, if it is None

### PR DESCRIPTION
**Pull request by dsavchenko**

### CODE REVIEW

**Comment by dsavchenko** [comment]
> Shall we apply the check also when generating the `job_id` and creating the `analysis_parameters.json` ?

I am not sure if the situation possible when some parameter being empty intentionally?

**Comment by burnout87** [comment]
> > Shall we apply the check also when generating the `job_id` and creating the `analysis_parameters.json` ?
> 
> I am not sure if the situation possible when some parameter being empty intentionally?

Exactly, probably is not necessary. It would just be an additional security check.

**Comment by dsavchenko** [comment]
Ok, empty parameter values in the request are being converted into empty string in the file, not null. Then it is safe t implement these additional 'is not None' checks

**Comment by burnout87** [comment]
> Ok, empty parameter values in the request are being converted into empty string in the file, not null. Then it is safe t implement these additional 'is not None' checks

In this way we are also 'consistent' with the way we treat the request parameters

**Review by burnout87 — APPROVED** [review]

File: cdci_data_analysis/flask_app/dispatcher_query.py [hunk 1]
```diff
@@ -232,7 +232,8 @@ def format_hash(x): return hashlib.md5(
         return format_hash(json.dumps(o))
 
     def calculate_job_id(self, par_dic):
-        return u'%s' % (self.make_hash(par_dic))
```
Comment (volodymyrss) [inline cdci_data_analysis/flask_app/dispatcher_query.py:235]: I am not too sure why we have both `make_hash` and `calculate_job_id` but ok, it's not part of this PR.

File: cdci_data_analysis/analysis/instrument.py [hunk 2]
```diff
@@ -551,7 +551,8 @@ def set_catalog_from_fronted(self,par_dic,request,back_end_query,logger=None,ver
             #print('POST')
             try:
                 cat_file_path = back_end_query.upload_file('user_catalog_file', back_end_query.scratch_dir)
-                par_dic['user_catalog_file'] = cat_file_path
+                if cat_file_path is not None:
```
Comment (volodymyrss) [inline cdci_data_analysis/analysis/instrument.py:554]: could there be a test? Cat uploading caused quite some problems, it should be tested. If you do not want, let's make an issue and we do soon after.

**Review by volodymyrss — CHANGES_REQUESTED** [review]

File: cdci_data_analysis/flask_app/dispatcher_query.py (hunk 1 above)
Comment (burnout87) [reply cdci_data_analysis/flask_app/dispatcher_query.py:235]: `calculate_job_id` is used when generating a `job_id` starting from the dictionary of the request, and during the vavlidation of the `job_id` (calculation from the `analysis_parameters.json` file)

File: cdci_data_analysis/analysis/instrument.py (hunk 2 above)
Comment (dsavchenko) [reply cdci_data_analysis/analysis/instrument.py:554]: I do not fully know this catalog functionality, and not aware of other issues. So anyway will need some clarifications to write meaningful tests. Thus better to split into another PR.

File: cdci_data_analysis/flask_app/dispatcher_query.py (hunk 1 above)
Comment (dsavchenko) [reply cdci_data_analysis/flask_app/dispatcher_query.py:235]: Another copy of the make_hash is in the pytests_fixtures

Comment (volodymyrss) [reply cdci_data_analysis/flask_app/dispatcher_query.py:235]: yes, very good.

File: cdci_data_analysis/analysis/instrument.py (hunk 2 above)
Comment (volodymyrss) [reply cdci_data_analysis/analysis/instrument.py:554]: ok #173

File: cdci_data_analysis/flask_app/dispatcher_query.py (hunk 1 above)
Comment (volodymyrss) [reply cdci_data_analysis/flask_app/dispatcher_query.py:235]: yes, very good.

**Review by volodymyrss — APPROVED** [review]
